org.testeditor.web.backend.persistence/src/test/java/org/testeditor/web/backend/testexecution/TestMonitorProviderTest.xtend

```diff
@@ -0,0 +1,223 @@
+		when(testProcess.exitValue).thenReturn(EXIT_SUCCESS)
+		when(testProcess.alive).thenReturn(false)
+		when(secondProcess.alive).thenReturn(true)
+		assertThat(testMonitorProviderUnderTest.getStatus(testPath)).isNotEqualTo(RUNNING)
+
+		// when
+		testMonitorProviderUnderTest.addTestRun(testPath, secondProcess)
+
+		// then
+		assertThat(testMonitorProviderUnderTest.getStatus(testPath)).isEqualTo(RUNNING)
+	}
+
+	@Test
+	def void getStatusReturnsIdleForUnknownTestPath() {
+		// given
+		val testPath = '/path/to/test.tcl'
+
+		// when
+		val actualStatus = testMonitorProviderUnderTest.getStatus(testPath)
+
+		// then
+		assertThat(actualStatus).isEqualTo(TestStatus.IDLE)
+	}
+
+	@Test
+	def void getStatusReturnsRunningAsLongAsTestProcessIsAlive() {
+		// given
+		val testProcess = mock(Process)
+		val testPath = '/path/to/test.tcl'
+		when(testProcess.alive).thenReturn(true)
+		when(testProcess.exitValue).thenThrow(new IllegalStateException("Process is still running"))
+		testMonitorProviderUnderTest.addTestRun(testPath, testProcess)
+
+		// when
+		val actualStatus = testMonitorProviderUnderTest.getStatus(testPath)
+
+		// then
+		assertThat(actualStatus).isEqualTo(TestStatus.RUNNING)
+	}
+
+	@Test
+	def void getStatusReturnsSuccessAfterTestFinishedSuccessfully() {
+		val testProcess = mock(Process)
+		val testPath = '/path/to/test.tcl'
+		when(testProcess.exitValue).thenReturn(EXIT_SUCCESS)
+		when(testProcess.waitFor).thenReturn(EXIT_SUCCESS)
+		testMonitorProviderUnderTest.addTestRun(testPath, testProcess)
+
+		// when
+		val actualStatus = testMonitorProviderUnderTest.getStatus(testPath)
+
+		// then
+		assertThat(actualStatus).isEqualTo(TestStatus.SUCCESS)
+	}
+
+	@Test
+	def void getStatusReturnsFailureAfterTestFailed() {
+		val testProcess = mock(Process)
+		val testPath = '/path/to/test.tcl'
+		when(testProcess.exitValue).thenReturn(EXIT_FAILURE)
+		when(testProcess.waitFor).thenReturn(EXIT_FAILURE)
+		testMonitorProviderUnderTest.addTestRun(testPath, testProcess)
+
+		// when
+		val actualStatus = testMonitorProviderUnderTest.getStatus(testPath)
+
+		// then
+		assertThat(actualStatus).isEqualTo(TestStatus.FAILED)
+	}
+
+	@Test
+	def void getStatusReturnsFailureWhenExternalProcessExitsWithNoneZeroCode() {
+		val testProcess = new ProcessBuilder('sh', '-c', '''exit «EXIT_FAILURE»''').start
+		val testPath = '/path/to/test.tcl'
+		testMonitorProviderUnderTest.addTestRun(testPath, testProcess)
+		testProcess.waitFor
+
+		// when
+		val actualStatus = testMonitorProviderUnderTest.getStatus(testPath)
+
+		// then
+		assertThat(actualStatus).isEqualTo(TestStatus.FAILED)
+	}
+
+	@Test
+	def void waitForStatusReturnsIdleForUnknownTestPath() {
+		// given
+		val testPath = '/path/to/test.tcl'
+
+		// when
+		val actualStatus = testMonitorProviderUnderTest.waitForStatus(testPath)
+
+		// then
+		assertThat(actualStatus).isEqualTo(TestStatus.IDLE)
+	}
+
+	@Test
+	def void waitForStatusBlocksUntilTestProcessTerminates() {
```

Comment:
this is not really tested, is it? It is tested however that processes will call 'waitFor' before the waitForStatus returns, right? (This should of course be semantically the same ...)

Reply:
Yeah, I tried to actually do wait (using `CountdownLatch`es), but the signalling would have been quite complicated, at least I did not find a good way to test it that way.

renamed to `waitForStatusCallsBlockingWaitForMethodOfProcess`